app/src/main/java/fr/free/nrw/commons/upload/mediaDetails/UploadMediaDetailFragment.java

```diff
@@ -284,7 +296,15 @@ public void onImageProcessed(UploadItem uploadItem, Place place) {
         }
 
         descriptions = uploadItem.getDescriptions();
-        photoViewBackgroundImage.setImageURI(uploadItem.getMediaUri());
+        compositeDisposable
+            .add(downSampleImage(uploadItem.getContentUri())
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(bitmap -> {
+                    if (null != bitmap) {
+                        photoViewBackgroundImage.setImageBitmap(bitmap);
```

Comment:
Can we use a SimpleDraweeView and `.setContentUri` to avoid manual bounds decoding?

Reply:
@macgills  made the required changes, there seems to be a small concern though, we used to use attach a scale change listener which is not there with the ZoomableDraweeView, I had to add a zoom control listener but as you can see I need to call invaliDate in the onTransformEnd to let the zoom to actually happen. Do you know some other method I could override to do this in a better way?